pkg/subctl/cmd/join.go

```diff
@@ -378,6 +383,21 @@ func determineServiceCIDR(nd *network.ClusterNetwork) error {
 	return nil
 }
 
+// determineExternalCIDR validates and appends the external cidr to cluster cidr
+func determineExternalCIDR(nd *network.ClusterNetwork) error {
+	if externalCIDR != "" {
+		err := utils.IsValidCIDR(externalCIDR)
+		if err == nil {
+			externalCIDR = utils.GetNetworkAddress(externalCIDR)
+			nd.PodCIDRs = append(nd.PodCIDRs, externalCIDR)
```

Comment:
This alone will not help because nd.PodCIDRs is mainly used to give a warning (line 357) to the user when the user-supplied information does not match with the auto-discovered values.

To get this working, please assign it back to clusterCIDR as shown below at L393
clusterCIDR = strings.Join(nd.PodCIDRs,",")

clusterCIDR is the one that gets added to the submarinerSpec in Line 497
When clusterCIDR is not explicitly specified, the submariner-operator will auto-discover the clusterCIDRs and use it when installing the Submariner Pods.

Reply:
Done, thanks Sridhar!